lib/ansible/plugins/callback/foreman.py

```diff
@@ -0,0 +1,246 @@
+        data = json.dumps(data)
+        facts_json = FACTS_FORMAT % dict(host=host, data=data)
+
+        requests.post(url=FOREMAN_URL + '/api/v2/hosts/facts',
+                      data=facts_json,
+                      headers=FOREMAN_HEADERS,
+                      cert=FOREMAN_SSL_CERT,
+                      verify=self.ssl_verify)
+
+    def _build_log(self, data):
+        logs = []
+        for entry in data:
+            if isinstance(entry, tuple):
+                # v2 plugins have the task name
+                source, msg = entry
+            else:
+                if 'invocation' in entry:
+                    source = json.dumps(entry['invocation'])
+                else:
+                    source = json.dumps({"encrypted": "true"})
+                msg = entry
+            if 'failed' in msg:
+                level = 'err'
+            else:
+                level = 'notice' if 'changed' in msg and msg['changed'] else 'info'
+            logs.append({"log": {
+                'sources': {'source': source},
+                'messages': {'message': json.dumps(msg)},
+                'level':     level
+                }})
+        return logs
+
+
+    def send_reports(self, stats):
+        """
+        Send reports to Foreman, to be parsed by Foreman config report
+        importer.  I massage the data get a report json that Foreman
+        can handle without writing another report importer.
+
+        Currently it just sets the status. It's missing:
+          - metrics, which we can get from data, except for runtime
+        """
+        status = defaultdict(lambda: 0)
+        metrics = {}
+
+        for host in stats.processed.keys():
+            sum = stats.summarize(host)
+            status["applied"] = sum['changed']
+            status["failed"] = sum['failures'] + sum['unreachable']
+            status["skipped"] = sum['skipped']
+            log = self._build_log(self.items[host])
+            metrics["time"] = {"total": int(time.time()) - self.start_time}
+            self.items[host] = []
+
+            report_json = REPORT_FORMAT % dict(host=host,
+                now=datetime.now().strftime(TIME_FORMAT),
+                metrics=json.dumps(metrics),
+                status=json.dumps(status),
+                log=json.dumps(log))
+            # To be changed to /api/v2/config_reports in 1.11.  Maybe we
+            # could make a GET request to get the Foreman version & do
+            # this automatically.
+            requests.post(url=FOREMAN_URL + '/api/v2/reports',
+                          data=report_json,
+                          headers=FOREMAN_HEADERS,
+                          cert=FOREMAN_SSL_CERT,
+                          verify=self.ssl_verify)
+
+    def on_any(self, *args, **kwargs):
+        pass
+
+    def runner_on_failed(self, host, res, ignore_errors=False):
+        self.items[host].append(res)
+
+    def runner_on_ok(self, host, res):
```

Comment:
this is never invoked as the v2_ method that would forward to this in the callbackbase is overridden below